doc/motivation.md

```diff
@@ -257,43 +257,42 @@ which will be resolved elsewhere in the RIB.
   </tr>
 </table>
 
-## Tying Injected Entries to the Liveliness of gRPC Client
-
-We expect that all entries that are installed by a gRPC client are only
-maintained during the time that a (long-lived) RPC remains open. In keeping
-with other control plane approaches, we expect that redundancy of entries can
-be maintained by having >1 source of injected entries. Where there is a
-requirement to persist entries that are to be installed within the gRPC service
-(e.g., static SR-TE LSPs), we expect that this is achieved through the
-management plane of the system.
-
-Since there is a long-lived RPC, we expect that there is a requirement to have
-a keepalive mechanism to ensure that the client is alive, although this could
-be achieved simply by tracking TCP session liveliness, it is possible that the
-gRPC
-[keepalive](https://github.com/grpc/grpc-go/blob/master/keepalive/keepalive.go)
-mechanism could be utilised, requiring little to no additional machinery to
-achieve such functionality. The requirement for such a keepalive is to
-determine a non-responsive client from one that is simply silent, such that in
-the case that >1 source of programming entries exists then stale client
-information can be determined.
-
-Since the possibility of having multiple clients introduces the possibility of
-requirements for tiebreaking, we propose that the implementation is kept as
-simple possible, and the client with the lowest client address when expressed
-in the form of an 128-bit number is selected. We expect that the remote systems
-are responsible for ensuring consistency of announcements.
-
+## Liveliness of Injected Entries
+
+There has been much debate about how entries installed by a routing protocol
+should be dealt with when the session, or neighbour, that provides those entries
+"goes away". Tension exists being approaches that prefer removing that routing
+information, and approaches that prefer to 'fail static' based on the last
+available entries. In order to provide for both approaches within gRIBI, two
+modes of operation are proposed.
+ 
+ * a mode of operation where the entries on the server are not tied to session
```

Comment:
Where should we specify the detailed spec of this mode (e.g. any timeout on the fail-static routes, how to deal with the fail-static route when client reconnect) ? 

Maybe a TODO here as place holder for further discussion/proposal?

Reply:
ACK - we should do this in the detailed spec I think. Currently, there is no timeout - an external client can call `Flush`, and these routes are essentially reconciled through `Get` plus `Modify` operations to bring the server in-line with the expected state.

I figured that just keeping this document extremely high-level such that it serves as "why gRIBI?" would be a good start, and have an outline of a spec that I was planning to push and see whether we should work on.

Let me know if this sounds reasonable -- or I can add something further here to describe this at a top-level if we think the doc is incomplete without it :-)

Reply:
+1 make sense.